graphium/data/dataset.py

```diff
@@ -152,178 +154,48 @@ class MultitaskDataset(Dataset):
 
     def __init__(
         self,
-        datasets: Dict[str, SingleTaskDataset],
-        n_jobs=-1,
-        backend: str = "loky",
-        featurization_batch_size=1000,
-        progress: bool = True,
-        save_smiles_and_ids: bool = False,
+        featurize_smiles: Callable[[str],dict],
+        task_names: List[str],
+        label_num_cols: List[int],
+        label_dtypes: List[int],
+        mol_file_data_offsets,
+        concat_smiles_tensor,
+        smiles_offsets_tensor,
+        num_nodes_tensor,
+        num_edges_tensor,
```

Comment:
I'm considering deleting `SingleTaskDataset`, because it's now only used by `FakeDataModule` and test_dataset.py, which needs reworking for the tests to pass again.

Reply:
I deleted `FakeDataset` and `SingleTaskDataset`.